src/ControlSaveTiles.js

```diff
@@ -225,14 +227,37 @@ const ControlSaveTiles = L.Control.extend(
     _loadTile: async function _loadTile(jtile) {
       const self = this;
       const tile = jtile;
-      await downloadTile(tile.url).then((blob) => {
-        self.status.lengthLoaded += 1;
-        self._saveTile(tile, blob);
-        self._baseLayer.fire('loadtileend', self.status);
-        if (self.status.lengthLoaded === self.status.lengthToBeSaved) {
-          self._baseLayer.fire('loadend', self.status);
-        }
-      });
+
+      if (this.options.alwaysDownload) {
+        await downloadTile(tile.url).then((blob) => {
+          self.status.lengthLoaded += 1;
```

Comment:
Thanks for your improvements to the code! I see 3 identical blocks here, can you make it one (eg add some method)